addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -32,12 +32,44 @@ options.registry.gallery = options.Class.extend({
             return layoutPromise.then(this._super.apply(this, arguments));
         }
 
-        // Make sure image previews are updated if images are changed
-        this.$target.on('image_changed.gallery', 'img', function (ev) {
-            var $img = $(ev.currentTarget);
-            var index = self.$target.find('.carousel-item.active').index();
-            self.$('.carousel:first li[data-bs-target]:eq(' + index + ')')
-                .css('background-image', 'url(' + $img.attr('src') + ')');
+        // Manage the media replacement. Make sure the data-index attribute and
+        // the media previews are updated if medias are changed.
+        // TODO init_media_dialog is not triggered anymore
+        this.$target.on("init_media_dialog.gallery", async (_, data) => {
+            // Prevent registering the replace media step. It will be registered
+            // manually once everything is updated, to have only one big step.
+            this.options.wysiwyg.odooEditor.historyPauseSteps();
+
+            // Save the media data-index to set it on the new one afterwards.
+            this.savedIndex = data.mediaEl.dataset.index;
+            // Only allow images and videos in the media dialog.
+            data.options.noIcons = true;
+            data.options.noDocuments = true;
+            // Make sure the steps are unpaused when closing the dialog.
+            data.options.onClose = () => this.options.wysiwyg.odooEditor.historyUnpauseSteps();
+        });
+        // TODO replace_target is not triggered anymore
+        this.$target.on("replace_target.gallery", "img, .media_iframe_video", async (_, mediaEl) => {
```

Comment:
I added the comment as a reminder, I don't think this event exist in 16.0. So if possible, find another way. If not possible, restore the event trigger? :shrug: